### ISSUE TITLE
minion id not working in zoo.cfg

### ISSUE THREAD

**Issue by hackerwin7** [saltstack-formulas/zookeeper-formula#54]
I use the grains 'roles: zookeeper', so after zookeeper.config, the zoo.cfg is like below:
`
server.1=minion1:2888:3888
server.2=minion2:2888:3888
server.3=minion3:2888:3888
`
It is not available to use zkCli.sh to connect to server
After I replace the minion id with the ip address, everything is OK!

**Comment by sroegner** [comment]
@hackerwin7 sounds to me your minion IDs are not hostnames/FQDNs that can be resolved on the host you're trying to use zkCli.sh on?

**Comment by hackerwin7** [comment]
Yes, key problem is that minion IDs are not hostnames or fqdn. but this is default settings envi for salt. 

Is that to say this zk formula require minions IDs are hostnames or fqdn which is not default settings in some salt envi ?

**Comment by hackerwin7** [comment]
Here is my envi:
3 minions:
```
127.0.0.1	saltminion3.local	saltminion3
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
```
Sorry, I find my key problem is that I explicitly set the minion id in /etc/salt/minion config file.

My three minions ip is 192.168.50.11 / 192.168.50.12 / 192.168.50.13
Is that to say I have to use grain or network models to get the specific ip address to apply into the pillar?

**Comment by hackerwin7** [comment]
One more problem, even we use default minion id 
The /etc/hosts is like this:
```
127.0.0.1	minion3
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
```
The minion id is just resolved to 127.0.0.1.
So I think the zoo.cfg below which is also unavailable in zookeeper:
```
server.1=minion1:2888:3888 
server.2=minion2:2888:3888 
server.3=minion3:2888:3888
```

**Comment by sroegner** [comment]
@hackerwin7 what would you expect the formula to do here? As far as i can tell your hosts have no IP address set in /etc/hosts (which doesn't mean they don't have one) but the default targeting mechanism of the formula does indeed require the minion id to be a resolvable name. Have you looked at (the bottom of) pillar.example - one of the alternative ways of targeting is described there.

**Comment by hackerwin7** [comment]
The  formula dose indeed require minion id resolvable but the default salt environment may be not compatible with this.
I'm using [this envi](https://github.com/UtahDave/salt-vagrant-demo) to apply the formula, however the formula is failed with the minion id which is not resolvable for this salt environment.
The section about hosts_function and mine_functions with network.ip_adddrs just work fine.
What confused me it that why using hostname or minion id or fqdn in default, I think it has no sense in a plain or wild salt environment, because the hostname or fqdn just resolve to 127.0.0.1.
This is so weird in zoo.cfg below:
```
server.1=minion1:2888:3888 
server.2=minion2:2888:3888 
server.3=minion3:2888:3888
```
the minion* just resolved to 127.0.0.1. and the salt envi is default to [salt-vagrant-demo](https://github.com/UtahDave/salt-vagrant-demo).

**Comment by codeboyQ2n5ha45** [comment]
Should be able to close this as formula has been refactored.
